wgpu-hal/src/auxil/dxgi/result.rs

```diff
@@ -1,37 +1,46 @@
 use std::borrow::Cow;
 
-use winapi::shared::winerror;
+use windows::Win32::{Foundation, Graphics::Dxgi};
 
 pub(crate) trait HResult<O> {
     fn into_result(self) -> Result<O, Cow<'static, str>>;
     fn into_device_result(self, description: &str) -> Result<O, crate::DeviceError>;
 }
-impl HResult<()> for i32 {
-    fn into_result(self) -> Result<(), Cow<'static, str>> {
-        if self >= 0 {
-            return Ok(());
-        }
+impl<T> HResult<T> for windows::core::Result<T> {
+    fn into_result(self) -> Result<T, Cow<'static, str>> {
+        // TODO: use windows-rs built-in error formatting?
```

Comment:
I think I forgot to look at simplifying this code...

Reply:
I'm planning to resolve this one in an upcoming PR.

Reply:
The built-in formatting for `Error::from_hresult(E_INVALIDARG)` is:

- `Display`: `The parameter is incorrect. (0x80070057)`
- `Debug`: `Error { code: HRESULT(0x80070057), message: "The parameter is incorrect." }`

So the former is already better than the hardcoded `invalid argument` string - and we cover more than just 4 errors.

I don't know why I didn't solve this one straight away :)

Reply:
I've killed `into_result()` entirely because the only uses either:

- Discard the message (which is weird anyway?)
- Print it in a `log::warn!()`, which could `Display`-format the `windows::core::Error` directly anyway.